src/transport_ofi.h

```diff
@@ -389,27 +412,55 @@ int shmem_transport_fence(shmem_transport_ctx_t* ctx)
 
 }
 
-/* RM requires polling until space is available */
+
+/* Process RMA operation return code.  If libfabric returned -FI_EAGAIN, attempt
+ * to reclaim resources and indicate that the operation should be retried.  If
+ * retry limit (ofi_max_poll) is exceeded, abort. */
 static inline
 int try_again(shmem_transport_ctx_t *ctx, const int ret, uint64_t *polled) {
 
     if (ret) {
         if (ret == -FI_EAGAIN) {
-            shmem_transport_ofi_drain_cq(ctx);
+            if (ctx->cq_ep) {
+                shmem_free_list_lock(ctx->bounce_buffers);
+                shmem_transport_ofi_drain_cq(ctx);
+                shmem_free_list_unlock(ctx->bounce_buffers);
+            }
+            else if (0) {
```

Comment:
Is this intentional?  I'd think `else {...` would be ok.

Reply:
Oh, good catch!  That's left over from debugging.